File: 1890-sum-of-beauty-of-all-substrings/1890-sum-of-beauty-of-all-substrings.cpp
class Solution {
public:
    typedef pair<char,int> p;
    int beautySum(string s) {
        int n = s.length();

        int ans = 0;

        for(int i = 0; i<n; i++){
            unordered_map<char,int> mp;
            for(int j = i; j < n ;j++){
                mp[s[j]]++;
                int mxFreq = 0, mnFreq = INT_MAX;
                for(auto &it : mp){
                    mxFreq = max(mxFreq, it.second);
                    mnFreq = min(mnFreq, it.second);
                }
                ans += mxFreq - mnFreq;
            }
            
        }
        return ans;
    }
};

File: 1890-sum-of-beauty-of-all-substrings/1890-sum-of-beauty-of-all-substrings.java
class Solution {
    public int beautySum(String s) {
        int n = s.length();

        int ans = 0;

        for(int i = 0; i<n; i++){
            int[] freq = new int[26];
            for(int j = i; j < n ;j++){
                freq[s.charAt(j)-'a']++;
                int mxFreq = 0, mnFreq = Integer.MAX_VALUE;
                for(int k = 0; k<26; k++){
                    mxFreq = Math.max(mxFreq, freq[k]);
                    if(freq[k] != 0)
                        mnFreq = Math.min(mnFreq, freq[k]);
                }
                ans += mxFreq - mnFreq;
            }
        }
        return ans;
    }
}